Sources/FoundationEssentials/Data/Data+Reading.swift

```diff
@@ -209,17 +209,17 @@ internal func readBytesFromFile(path inPath: PathOrURL, reportProgress: Bool, ma
     }
     
     let fileSize = min(Int(clamping: filestat.st_size), maxLength ?? Int.max)
-    let fileType = filestat.st_mode & S_IFMT
+    let fileType = mode_t(filestat.st_mode) & mode_t(S_IFMT)
     let preferredChunkSize = filestat.st_blksize
 #if !NO_FILESYSTEM
     let shouldMap = shouldMapFileDescriptor(fd, path: inPath, options: options)
 #else
     let shouldMap = false
 #endif
         
-    if fileType != S_IFREG {
+    if fileType != mode_t(S_IFREG) {
```

Comment:
For my reference, could you elaborate a bit more on why these casts are all necessary? What type is `S_IFREG` on Windows? Does it use some other integer type, or does it use `int`? If so I think I'd expect that declaring a `typealias mode_t = Int` (or the appropriate integer type) would eliminate the need for these casts because `S_IFREG` would be considered the same type was `fileType` here. Or is there a detail I might be missing here?

Reply:
There is no `mode_t` on Windows, there is an alias that I have created which is `int`. However, the type system often sees the values as `UInt16` which would then be incompatible with `int`. The explicit casting avoids the mismatched types on the various sides.